File: src/main/java/info/skyblond/archivedag/repo/GroupMetaRepository.java
package info.skyblond.archivedag.repo;

import info.skyblond.archivedag.model.entity.GroupMetaEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;

public interface GroupMetaRepository extends JpaRepository<GroupMetaEntity, String> {
    boolean existsByGroupName(String groupName);

    boolean existsByGroupNameAndOwner(String groupName, String owner);

    @Modifying
    void deleteByGroupName(String groupName);

    GroupMetaEntity findByGroupName(String groupName);

    @Modifying
    @Query("update GroupMetaEntity m set m.owner = ?2 where m.groupName = ?1")
    void updateGroupOwner(String groupName, String owner);

    Page<GroupMetaEntity> findAllByOwner(String owner, Pageable pageable);


    Page<GroupMetaEntity> findAllByGroupNameContains(String keyword, Pageable pageable);
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arstue/repo/GroupMetaRepository.kt
package info.skyblond.archivedag.arstue.repo

import info.skyblond.archivedag.arstue.entity.GroupMetaEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query

interface GroupMetaRepository : JpaRepository<GroupMetaEntity, String> {
    fun existsByGroupName(groupName: String): Boolean

    fun existsByGroupNameAndOwner(groupName: String, owner: String): Boolean

    @Modifying
    fun deleteByGroupName(groupName: String)

    fun findByGroupName(groupName: String): GroupMetaEntity?

    @Modifying
    @Query("update GroupMetaEntity m set m.owner = ?2 where m.groupName = ?1")
    fun updateGroupOwner(groupName: String, owner: String)

    fun findAllByOwner(owner: String, pageable: Pageable): Page<GroupMetaEntity>


    fun findAllByGroupNameContains(keyword: String, pageable: Pageable): Page<GroupMetaEntity>
}
